drivers/hmis_supplemental/app/jobs/hmis_supplemental/import_job.rb

```diff
@@ -0,0 +1,131 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class HmisSupplemental::ImportJob
+  attr_reader :data_set
+
+  # csv_string is mostly for QA
+  def perform(data_set_id:, csv_string: nil)
+    @data_set = HmisSupplemental::DataSet.where(id: data_set_id).first
+    return unless @data_set
+
+    csv_string ||= read_csv_from_s3
+
+    with_lock do
+      values = read_csv_rows(csv_string).
+        flat_map { |row| process_row(row) }.
+        compact
+
+      if values.empty?
+        log('Could not map custom data elements from CSV')
+        return
+      end
+
+      values = deduplicate_rows(values)
+      data_set.transaction do
+        # Delete and recreate. This could be an upsert
+        data_set.field_values.delete_all(:delete_all)
+        HmisSupplemental::FieldValue.import!(values, validate: false)
```

Comment:
Do you think we should add some configuration to determine if this is a delta or a full-refresh?  I'm fine with assuming full-refresh for now, but we might want to make that an explicit choice (via argument).

Reply:
Feels like support for incremental updates could come later? Maybe we will need to support it if the data sets are large. I'm unsure as to the scale

Reply:
We can kick this can for sure.